tests/test_iotedgedev_layereddeployments.py

```diff
@@ -36,3 +42,42 @@ def test_build_and_push_with_no_modules():
 
     assert 'ERROR' not in result.output
     assert 'bar-1.2' == set_property
+
+
+def test_deploy():
+    try:
+        # Arrange
+        deployment_name = f'test-{uuid.uuid4()}'
+        output = Output()
+        envvars = EnvVars(output)
+
+        os.chdir(test_solution_shared_lib_dir)
+        envvars.set_envvar("LAYERED_DEPLOYMENT_TARGET_CONDITION", "tags.environment='dev'")
+
+        # Act
+        result = runner_invoke(['build', '--push', '-f', "layered_deployment.template_with_flattened_props.json", '-P', get_platform_type()])
+        result = runner_invoke(['deploy-layered',
+                                '-f', 'config/layered_deployment.template_with_flattened_props.json',
+                                '-n', deployment_name,
+                                '-p', '10',
+                                ])
+
+        # Assert
+        assert 'DEPLOYMENT COMPLETE' in result.output
+        assert 'ERROR' not in result.output
+    finally:
+        # Cleanup
+        del os.environ["LAYERED_DEPLOYMENT_TARGET_CONDITION"]
+        azure_cli = AzureCli(output, envvars)
+        envvars.load()
+
+        assert azure_cli.invoke_az_cli_outproc(["iot", "edge", "deployment", "delete", "-d", deployment_name, "-l", envvars.IOTHUB_CONNECTION_INFO.connection_string])
+
+
+# TODO: Add tests cases:
+# - Verify error message is in the logs
+# - Verify LAYERED_DEPLOYMENT_TARGET_CONDITION cant be not set
+# - Verify required parameters can't be not set
+# - Cover use cases from build/push (test_build_and_push_with_no_modules)
+# - Check if a layered deployment flag works
+# -
```

Comment:
check for the case where the config file is only set in env and not specified through cmd line; also check for when env and cmd line both have config file